src/main/kotlin/org/rust/ide/utils/CallInfo.kt

```diff
@@ -22,14 +30,33 @@ class CallInfo private constructor(
     companion object {
         fun resolve(call: RsCallExpr): CallInfo? {
             val fn = (call.expr as? RsPathExpr)?.path?.reference?.resolve() ?: return null
-            if (fn is RsFunction) return CallInfo(fn)
-
             val ty = call.expr.type
-            if (ty is TyFunction) {
-                val parameterNames = getParameterNames(fn) ?: List(ty.paramTypes.size) { "_" }
-                return CallInfo(ty, parameterNames)
+
+            return when {
+                fn is RsFunction -> {
```

Comment:
Consider extracting branches bodies into private functions. Maybe something like this (if I understand the whole logic right):
```Kotlin
return when {
    fn is RsFunction -> handleFunction(fn, ty)
    ty is TyFunction -> handleEnum(fn, ty)
    else -> null
}
```